src/controller/CommissioneeDeviceProxy.h

```diff
@@ -216,7 +216,7 @@ class CommissioneeDeviceProxy : public DeviceProxy, public SessionReleaseDelegat
 
     Messaging::ExchangeManager * GetExchangeManager() const override { return mExchangeMgr; }
 
-    void SetAddress(const Inet::IPAddress & deviceAddr) { mDeviceAddress.SetIPAddress(deviceAddr); }
+    void SetAddress(const Inet::IPAddress & deviceAddr) { mDeviceAddress.SetSingleIPAddress(deviceAddr); }
```

Comment:
This is the suspect case: SetAddress ignores what interface mDeviceAddress has.
So if this address is link-local, it is probably wrong, unless we have a SetInterface. We will need to trace what calls this and how and convert into some 'SetSingleDestination' or 'SetPeerAddress' instead.

Reply:
If you delete this method, what breaks? It 100% is a bad thing to ignore interface and we should not have any APIs that do so related to any discovery

Reply:
I want to do that as a followup. I do not know what a 'Commissionee Device proxy' is ... I imagine I will have to track down the usage and at some point I should end up at a DNSSD lookup or some manual input somehow.